src/nanoemoji/paint.py

```diff
@@ -299,6 +312,233 @@ def to_ufo_paint(self, colors):
+    format: ClassVar[int] = int(ot.PaintFormat.PaintScaleUniformAroundCenter)
+    paint: Paint
+    scale: float = 1.0
+    center: Point = Point()
+
+    def colors(self):
+        yield from self.paint.colors()
+
+    def to_ufo_paint(self, colors):
+        paint = {
+            "Format": self.format,
+            "Paint": self.paint.to_ufo_paint(colors),
+            "scale": self.scale,
+            "centerX": self.center[0],
+            "centerY": self.center[1],
+        }
+        return paint
+
+    def children(self) -> Iterable[Paint]:
+        return (self.paint,)
+
+    def gettransform(self) -> Affine2D:
+        return (
+            Affine2D.identity()
+            .translate(self.center[0], self.center[1])
+            .scale(self.scale)
+            .translate(-self.center[0], -self.center[1])
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class PaintRotate(Paint):
+    format: ClassVar[int] = int(ot.PaintFormat.PaintRotate)
+    paint: Paint
+    angle: float = 0.0
+
+    def colors(self):
+        yield from self.paint.colors()
+
+    def to_ufo_paint(self, colors):
+        paint = {
+            "Format": self.format,
+            "Paint": self.paint.to_ufo_paint(colors),
+            "angle": self.angle,
+        }
+        return paint
+
+    def children(self) -> Iterable[Paint]:
+        return (self.paint,)
+
+    def gettransform(self) -> Affine2D:
+        return Affine2D.identity().rotate(radians(self.angle))
+
+
+@dataclasses.dataclass(frozen=True)
+class PaintRotateAroundCenter(Paint):
+    format: ClassVar[int] = int(ot.PaintFormat.PaintRotateAroundCenter)
+    paint: Paint
+    angle: float = 0.0
+    center: Point = Point()
+
+    def colors(self):
+        yield from self.paint.colors()
+
+    def to_ufo_paint(self, colors):
+        paint = {
+            "Format": self.format,
+            "Paint": self.paint.to_ufo_paint(colors),
+            "angle": self.angle,
+            "centerX": self.center[0],
+            "centerY": self.center[1],
+        }
+        return paint
+
+    def children(self) -> Iterable[Paint]:
+        return (self.paint,)
+
+    def gettransform(self) -> Affine2D:
+        return Affine2D.identity().rotate(
+            radians(self.angle), self.center[0], self.center[1]
+        )
+
+
+@dataclasses.dataclass(frozen=True)
+class PaintSkew(Paint):
+    format: ClassVar[int] = int(ot.PaintFormat.PaintSkew)
+    paint: Paint
+    xSkewAngle: float = 0.0
+    ySkewAngle: float = 0.0
+
+    def colors(self):
+        yield from self.paint.colors()
+
+    def to_ufo_paint(self, colors):
+        paint = {
+            "Format": self.format,
+            "Paint": self.paint.to_ufo_paint(colors),
+            "xSkewAngle": self.xSkewAngle,
+            "ySkewAngle": self.ySkewAngle,
+        }
+        return paint
+
+    def children(self) -> Iterable[Paint]:
+        return (self.paint,)
+
+    def gettransform(self) -> Affine2D:
+        raise NotImplementedError()
```

Comment:
why NotImplementedError? Affine2D does have `skewx` and `skewy` methods that work on degree angles

Reply:
Added implementation